applications/IgaApplication/custom_modelers/nurbs_geometry_modeler.cpp

```diff
@@ -152,8 +152,14 @@ namespace Kratos
             insert_knots_v[i] = knot_v;
         }
 
-        const SizeType number_of_geometries = r_model_part.NumberOfGeometries();
-        p_surface_geometry->SetId(number_of_geometries+1);
+        // Add geometry to model part
+        const SizeType number_geometries = r_model_part.NumberOfGeometries();
+        SizeType last_geometry_id = 0;
+        if( number_geometries > 0 ){
+            last_geometry_id = r_model_part.Geometries()[number_geometries-1].Id();
+        }
```

Comment:
```suggestion
        SizeType last_geometry_id = ( r_model_part.NumberOfGeometries() > 0 )
            ? r_model_part.Geometries().back().Id()
            : 0;
```

The geometry container does not work like this as it is a hashtable. You need to access the required id through the pointer to the last entry.